File: archive/v2-csharp-avalonia/src/Spanova.Core/Model/SpanLayout.cs
namespace Spanova.Core.Model;

/// <summary>An ordered sequence of spans along the bridge alignment.</summary>
public sealed class SpanLayout
{
    public IReadOnlyList<Span> Spans { get; init; } = Array.Empty<Span>();

    public int SpanCount => Spans.Count;

    /// <summary>Sum of all span lengths, in meters.</summary>
    public double TotalLengthM => Spans.Sum(s => s.LengthM);
}

File: backend/bridge-core/src/main/java/com/kopruq/bridgecore/SpanLayout.java
package com.kopruq.bridgecore;

import java.util.List;

/** An ordered sequence of spans along the bridge alignment. */
public record SpanLayout(List<Span> spans) {

    public int spanCount() {
        return spans.size();
    }

    /** Sum of all span lengths, in meters. */
    public double totalLengthM() {
        return spans.stream().mapToDouble(Span::lengthM).sum();
    }
}
